src/Glpi/Controller/Session/ChangeProfileController.php

```diff
@@ -73,8 +73,10 @@ public function __invoke(Request $request): Response
             $route = $go_to_create_ticket ? "/ServiceCatalog" : "/Helpdesk";
             $redirect = $request->getBasePath() . $route;
         } else {
-            $_SESSION['_redirected_from_profile_selector'] = true;
             $redirect = Html::getBackUrl();
+            $has_query_string = parse_url($redirect, PHP_URL_QUERY) !== null;
+            $separator = $has_query_string ? "&" : "?";
```

Comment:
Should not we check for `?` presence in the URL rather than using `parse_url()`? 
As far as I've heard, it's not always totally reliable (no idea for that specific part), so it's maybe useless here.